bundles/org.openhab.binding.myuplink/src/main/resources/OH-INF/thing/readonly-channel-types.xml

```diff
@@ -0,0 +1,77 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="myuplink"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+
+	<channel-type id="type-number-integer" advanced="true">
+		<item-type>Number</item-type>
+		<label>Generic Number</label>
+		<state pattern="%d" readOnly="true">
+		</state>
+	</channel-type>
+	<channel-type id="type-number-double" advanced="true">
+		<item-type>Number</item-type>
+		<label>Generic Number (0.1)</label>
+		<state pattern="%.1f" readOnly="true">
+		</state>
+	</channel-type>
+	<channel-type id="type-temperature">
+		<item-type>Number:Temperature</item-type>
+		<label>Generic Temperature</label>
+		<state pattern="%.1f %unit%" readOnly="true">
+		</state>
+	</channel-type>
+	<channel-type id="type-flow">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Generic Flow</label>
+		<state pattern="%.1f l/min" readOnly="true">
+		</state>
+	</channel-type>
```

Comment:
UoM should be used for this: Number:VolumetricFlowRate
For reference: https://www.openhab.org/docs/concepts/units-of-measurement.html#list-of-units

Reply:
adopted